torch/_inductor/codecache.py

```diff
@@ -279,7 +282,31 @@ def process_pool():
         # are forked
         cuda_properties._properties()
         assert config.compile_threads > 1
-        return ProcessPoolExecutor(config.compile_threads)
+        orig_ppid = os.getpid()
+
+        # if this process dies abnormally (e.g. segfault)
+        # it will not shut down the workers. Instead
+        # the workers will have their parent reassigned to the
+        # init process. This launches a separate thread to
+        # watch for the worker getting reassigned,
+        # and cleans it up in this case.
+        def init():
+            def run():
+                while True:
+                    if orig_ppid != os.getppid():
```

Comment:
a big try/catch around this to just ignore any exceptions that arise (probably during the main process's shutdown)?

Reply:
What sequence of events could arise that catching an exception will resolve? os.getpid and os.getppid can't throw exceptions,  the underlying unix calls do not error. os.kill can, but if we reach that code, we want the current process to die, and if the kill command doesn't work, we don't have a way to kill the process anyway.

Reply:
idk, i guess the comment was just to be preemptive. Didn't really think too much about it.